modules/history/src/views/full/index.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+import style from './style.scss'
+import moment from 'moment'
+
+import { MessagesViewer } from './message'
+import { ConversationPicker } from './conversation'
+
+export default class FullView extends React.Component {
+  constructor(props) {
+    super(props)
+
+    const blob = new Blob([''], { type: 'application/json' })
+    const url = window.URL.createObjectURL(blob)
+
+    const defaultToDate = moment().startOf('day')
+
+    const defaultFromDate = moment()
+      .startOf('day')
+      .subtract(30, 'days')
+
+    this.state = {
+      conversationsInfo: [],
+      messageGroups: [],
+      to: defaultToDate,
+      from: defaultFromDate,
+      currentConvId: null,
+      fileBlob: blob,
+      fileURL: url
+    }
+  }
+
+  threadIdParamName = 'threadId'
+
+  componentDidMount() {
+    this.getConversations(this.state.from, this.state.to)
+    const url = new URL(window.location.href)
+    const threadId = url.searchParams.get(this.threadIdParamName)
+    if (threadId) {
+      this.setState({ currentConvId: threadId })
+      this.getMessagesOfConversation(threadId)
+    }
+  }
+
+  getConversations(from, to) {
+    const ceiledToDate = moment(to).add(1, 'days')
+
+    this.props.bp.axios
+      .get(`/mod/history/conversations?from=${from.unix()}&to=${ceiledToDate.unix()}`)
+      .then(({ data }) => {
+        this.setState({ conversationsInfo: data })
+      })
+  }
+
+  onConversationSelected(convId) {
```

Comment:
The name would make more sense like "selectConversation(convoId)" , since it is used to fetch it, and not triggered when it is changed